app/screens/select_server/select_server.js

```diff
@@ -80,12 +82,19 @@ export default class SelectServer extends PureComponent {
             connected: false,
             connecting: false,
             error: null,
-            url: props.serverUrl,
         };
 
         this.cancelPing = null;
     }
 
+    static getDerivedStateFromProps(props, state) {
```

Comment:
do we really need this? is this being done like this cause of the `deepLinkURL` prop is set after this component mounts? Also the serverUrl prop is not being used to set in the `url` state, so the initial value if exists (like previous connected server url and/or emm server configuration is not being applied.

Reply:
Just detected a bug on this one though, it's fixed with my latest commit. Actually the statement`if !state.url ...` made the input value unchangeable once it is set with the deeplink URL. When input value was manually erased `state.url` became an empty string so the deeplink is recursively reset as the input value.

Reply:
this still is not taking into account `props.serverUrl` which was removed from the constructor